src/services/run-tests.ts

```diff
@@ -6,12 +7,24 @@ import { availableParallelism } from '../polyfills/os.js';
 import { hr, log } from '../services/write.js';
 import { format } from './format.js';
 import { runTestFile } from './run-test-file.js';
+import {
+  executeResourceFiles,
+  separateResourceFiles,
+} from './shared-resource-loader.js';
 
 const { cwd } = GLOBAL;
 
 if (hasOnly) deepOptions.push('--only');
 
 export const runTests = async (files: string[]): Promise<boolean> => {
+  const { resourceFiles, testFiles } = separateResourceFiles(files);
+
+  const registry: Record<string, SharedResourceEntry> = {};
```

Comment:
A quick tip: when creating an empty object or an object with properties that can be set dynamically, use `Object.create(null)` to avoid prototype pollution 🙋🏻‍♂️

```suggestion
  const registry: Record<string, SharedResourceEntry> = Object.create(null);
```